// @botpress/builtins/dist/renderers/builtin_image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _url = require('url');

var _url2 = _interopRequireDefault(_url);

var _mime = require('mime');

var _mime2 = _interopRequireDefault(_mime);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// TODO
// Add image support to Twilio (SMS)

exports.default = data => [{
  on: 'facebook',
  image: _url2.default.resolve(data.BOT_URL, data.image),
  typing: data.typing
}, {
  on: 'webchat',
  type: 'file',
  url: _url2.default.resolve(data.BOT_URL, data.image),
  typing: data.typing
}, {
  on: 'microsoft',
  attachments: [{
    contentType: _mime2.default.getType(data.image),
    contentUrl: data.image,
    name: data.title
  }]
}, {
  on: 'slack',
  attachments: [{
    title: data.title,
    image_url: data.image
  }]
}, {
  on: 'telegram',
  photo: _url2.default.resolve(data.BOT_URL, data.image),
  options: {
    caption: data.title,
    parse_mode: 'Markdown'
  }
}, {
  on: 'twilio',
  twilioBuiltinImage: _url2.default.resolve(data.BOT_URL, data.image),
  image: _url2.default.resolve(data.BOT_URL, data.image)
}];
